pkgs/applications/science/math/ratpoints/default.nix

```diff
@@ -1,20 +1,34 @@
-{stdenv, fetchurl, gmp}:
+{ stdenv, fetchurl, fetchpatch, gmp }:
 stdenv.mkDerivation rec {
   name = "ratpoints-${version}";
-  version = "2.1.3";
+  version = "2.1.3.p4";
+
   src = fetchurl {
     url = "http://www.mathe2.uni-bayreuth.de/stoll/programs/ratpoints-${version}.tar.gz";
     sha256 = "0zhad84sfds7izyksbqjmwpfw4rvyqk63yzdjd3ysd32zss5bgf4";
   };
-  buildInputs = [gmp];
-  makeFlags = "INSTALL_DIR=$(out)";
+
+  patches = [
+    (fetchpatch {
+      url = "https://git.sagemath.org/sage.git/plain/build/pkgs/ratpoints/patches/sturm_and_rp_private.patch";
```

Comment:
Can your refer to a specific commit? Otherwise an update of the patch will break the hash.